src/shared/dataFetcher.js

```diff
@@ -0,0 +1,12 @@
+import config from '../config/config';
+
+const dataFetcher = (url, method, token, params) => fetch(`${config.url}${url}`, {
```

Comment:
The data fetcher should get the token from the local storage by itself. So it does not need to be passed through the store and components.